### DISCUSSION TITLE
Can't create sqlite storage on cluster

### DISCUSSION THREAD

**Discussion by AHolliday** [optuna/optuna#4294] [Q&A] [answered]
Using optuna 3.0.5, I've been trying to run a hyperparameter optimization on a neural network model on a cluster I have access to.  I know we're discouraged from using SQLite for this but I haven't been given access to the MySQL or postreSQL servers on the cluster yet so I'm stuck with SQLite for the moment.  Things were running alright for a few days until yesterday, when the two jobs I had running both failed due to a locked-database issue (which had happened many times before).  I tried to start them again but found that optuna's create_study() was now raising an AssertionError at line 1161 of storages/_rdb/storage.py.  I deleted the existing .db file, but this didn't make any difference.  I now find that any attempt to create an RDBStorage object causes this error to occur, as in the minimal example below:

```
Python 3.8.10 (default, Jun 16 2021, 14:19:02) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import optuna
>>> storage = optuna.storages.RDBStorage("sqlite:///test.db")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lustre03/project/6005607/ahollid/ptlearn/lib/python3.8/site-packages/optuna-3.0.5-py3.8.egg/optuna/storages/_rdb/storage.py", line 225, in __init__
    self._version_manager.check_table_schema_compatibility()
  File "/lustre03/project/6005607/ahollid/ptlearn/lib/python3.8/site-packages/optuna-3.0.5-py3.8.egg/optuna/storages/_rdb/storage.py", line 1134, in check_table_schema_compatibility
    current_version = self.get_current_version()
  File "/lustre03/project/6005607/ahollid/ptlearn/lib/python3.8/site-packages/optuna-3.0.5-py3.8.egg/optuna/storages/_rdb/storage.py", line 1161, in get_current_version
    assert version is not None
AssertionError
```
Also, the call to RDBStorage() or create_study() (which internally calls the RDBStorage constructor) is now extremely slow.  It takes several minutes to come back with the AssertionError.  Note that 'test.db' did not exist prior to running the above example.  The same constructor works fine on another machine.

My first guess is that some kind of bad directory or file was left behind by an earlier job which is now being read and stopping optuna from creating the storage object, but I can't seem to find it.  Can anyone help me?

**Comment by nzw0301** [comment, marked as answer]
In terms of `AssertionError`, I have a similar error with `SQLAlchemy==2.0.0rc1`; downgrading it from `2.0.0rc1` to `1.4.46` resolved the error in my case.

**Reply by AHolliday** [reply]
That fixed it for me.  Strange.  Thanks.

**Reply by AntonClaesson** [reply]
Seems as it is an incompatibility issue with SQLAlchemy 2.0, which seem to be supported for Optuna 3.1.0
That's why downgrading to SQLAlchemy < 2.0 works
